### PR TITLE
Add BlockchainListener integration test

### LINKED ISSUE #1: Add gevent based blockchain event listener
- [x] BlockchainListener implementation
- [x] BlockchainListenerMock implementation
- [ ] Integration tests

### PR DESCRIPTION
**Pull request by palango**
This also reorganizes the tests:
- The raiden_libs.test module contains code that might be useful for
  other projects, e.g. text fixtures, mock classes...
- The tests module contains test code for raiden_libs itself.

This also fixes a small oversight in the BlockchainListener itself.

Also fixes #1

### CODE REVIEW

File: requirements_dev.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-pip==9.0.1
+pip==9.0.3
```
Comment (pcppcp) [inline requirements_dev.txt]: do we need forced versions?

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from raiden_libs.test.fixtures.address import *  # noqa
+from raiden_libs.test.fixtures.contracts import *  # noqa
+from raiden_libs.test.fixtures.client import *  # noqa
+from raiden_libs.test.fixtures.web3 import *  # noqa
+
+from .fixture_overwrites import *  # noqa
+
+
+def pytest_addoption(parser):
```
Comment (pcppcp) [inline tests/conftest.py]: let's assume tests work with `eth-tester` only, until proved otherwise

File: tests/fixture_overwrites.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+import pytest
+from raiden_contracts.contract_manager import ContractManager, CONTRACTS_SOURCE_DIRS
+from raiden_libs.blockchain import BlockchainListener
+
+
+@pytest.fixture(scope='session')
```
Comment (pcppcp) [inline tests/fixture_overwrites.py]: we can even have some tests disabled conditionally, depending on availability of libraries. See
https://github.com/raiden-network/microraiden/blob/master/microraiden/test/conftest.py#L14

File: tests/conftest.py (hunk 2 above)
Comment (palango) [reply tests/conftest.py]: Ok, I'll remove this

File: requirements_dev.txt (hunk 1 above)
Comment (palango) [reply requirements_dev.txt]: Don't think so. Will fix this.

File: tests/fixture_overwrites.py (hunk 3 above)
Comment (palango) [reply tests/fixture_overwrites.py]: Did you have anything special in mind?

**Comment by palango** [comment]
This turned out to be a bunch of changes. Please check again @pcppcp

File: tests/fixture_overwrites.py (hunk 3 above)
Comment (pcppcp) [reply tests/fixture_overwrites.py]: I mean the tests that require compilation - if `solc` or `raiden-contracts` is unavailable, they can be skipped. Not sure if that's required, probably not.

File: requirements_dev.txt [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
-tox==2.9.1
-coverage==4.5.1
-Sphinx==1.7.0
-sphinx_rtd_theme==0.2.4
-
-pytest==3.4.2
-pytest-runner==2.11.1
+-r requirements.txt
+
+pip>=9.0.1,<10.0
```
Comment (pcppcp) [inline requirements_dev.txt]: do you generate the list with some tool? I would personally keep the list without versions, and use frozen version only if required for compatibility reasons.

File: tests/fixture_overwrites.py (hunk 3 above)
Comment (palango) [reply tests/fixture_overwrites.py]: Lets add this later or if it goes out hand

**Pull request merged**